pkg/externalLink/ExternalLinkService.go

```diff
@@ -0,0 +1,341 @@
+		if err != nil && pg.ErrNoRows != err {
+			impl.logger.Errorw("error in fetch links by cluster id", "err", err)
+			return nil, err
+		}
+	}
+	var externalLinkResponse []*ExternalLinkDto
+	response := make(map[int]*ExternalLinkDto)
+	for _, link := range allActiveExternalLinkMapping {
+		if _, ok := response[link.ExternalLinkId]; !ok {
+			response[link.ExternalLinkId] = &ExternalLinkDto{
+				Id:               link.ExternalLinkId,
+				Name:             link.ExternalLink.Name,
+				Url:              link.ExternalLink.Url,
+				Active:           link.ExternalLink.Active,
+				MonitoringToolId: link.ExternalLink.ExternalLinkMonitoringToolId,
+				UpdatedOn:        link.UpdatedOn,
+			}
+		}
+		response[link.ExternalLinkId].ClusterIds = append(response[link.ExternalLinkId].ClusterIds, link.ClusterId)
+	}
+
+	for _, v := range response {
+		externalLinkResponse = append(externalLinkResponse, v)
+	}
+
+	//now add all the links which are not mapped to any clusters
+	additionalExternalLinks, err := impl.externalLinkRepository.FindAllFilterOutByIds(mappedExternalLinksIds)
+	if err != nil && pg.ErrNoRows != err {
+		impl.logger.Errorw("error in fetch all links", "err", err)
+		return nil, err
+	}
+	for _, link := range additionalExternalLinks {
+		providerRes := &ExternalLinkDto{
+			Id:               link.Id,
+			Name:             link.Name,
+			Url:              link.Url,
+			Active:           link.Active,
+			MonitoringToolId: link.ExternalLinkMonitoringToolId,
+			ClusterIds:       []int{},
+			UpdatedOn:        link.UpdatedOn,
+		}
+		externalLinkResponse = append(externalLinkResponse, providerRes)
+	}
+	if externalLinkResponse == nil {
+		externalLinkResponse = make([]*ExternalLinkDto, 0)
+	}
+	return externalLinkResponse, err
+}
+func (impl ExternalLinkServiceImpl) Update(request *ExternalLinkDto) (*ExternalLinkApiResponse, error) {
+	impl.logger.Debugw("link update request", "req", request)
+	dbConnection := impl.externalLinkRepository.GetConnection()
+	tx, err := dbConnection.Begin()
+	if err != nil {
+		impl.logger.Errorw("error in establishing connection", "err", err)
+		return nil, err
+	}
+	// Rollback tx on error.
+	defer tx.Rollback()
+	externalLink, err0 := impl.externalLinkRepository.FindOne(request.Id)
+	if err0 != nil {
+		impl.logger.Errorw("No matching entry found for update.", "id", request.Id)
+		return nil, err0
+	}
+	externalLink.Name = request.Name
+	externalLink.Url = request.Url
+	externalLink.Active = true
+	externalLink.ExternalLinkMonitoringToolId = request.MonitoringToolId
+	externalLink.UpdatedBy = int32(request.UserId)
+	externalLink.UpdatedOn = time.Now()
+	err = impl.externalLinkRepository.Update(&externalLink, tx)
+	if err != nil {
+		impl.logger.Errorw("error in updating link", "data", externalLink, "err", err)
+		return nil, err
+	}
+
+	allExternalLinksMapping, err := impl.externalLinkClusterMappingRepository.FindAllByExternalLinkId(request.Id)
+	if err != nil {
```

Comment:
should we handle noRowError?

Reply:
this handle is enough, we need to throw error here